src/GovUk.Education.ExploreEducationStatistics.Admin/appsettings.json

```
@@ -28,6 +28,11 @@
     "TokenValidationParameters.NameClaimType": "name",
     "TokenValidationParameters.RoleClaimType": "role"
   },
+  "OpenIdConnectSpaClient": {
+    "AllowOfflineAccess": true,
```

Should `AllowOfflineAccess` here be `false`?

```suggestion
    "AllowOfflineAccess": false,
```

I was going to say maybe don't even include this block here so there's no confusion over whether these settings might be taking effect over the settings defined in the App Service configuration in Azure, but I guess you've defined them here to provide a reference of all the possible settings?

---

Not really, so happy to remove :)  

Blatted.  I'm always confused by this file.  Presumably Dev uses some of this if it doesn't have override values specified in the ARM template?  Always confused as to why we have the Hive AD settings in this file rather than just appsettings.Development.json but always assumed that this was just inherited by Dev and that was the onlyreason to have it in here?

---

Yes that's correct, the settings in this `appsettings.json` file will take effect unless they are overridden by environment variables in the app service, (which are maintained by the infrastructure build through the ARM template).

In terms of precedence:

* command line options (highest)
* environment variables
* appsettings.{Environment}.json
* appsettings.json (lowest)

The three built-in environment names are `Development`, `Staging` and `Production`. All of our Azure environments run in `Production` and our local environment runs in `Development`.

So in our local environment, `appsettings.Development.json` has precedence over `appsettings.json`.

---

Great info, thanks for the reply!